elastic/apm-server
# docker: fix file access for apm-server user

## Motivation/summary

Ensure the apm-server.yml is readable by the apm-server user (UID 1000), and the `data/` directory is writeable by the same user.

This enables users to run the image with a non-root GID. This was possible in older versions, as the apm-server user was created with the root supplementary group.

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Create a local Kubernetes cluster using Kind
2. Install ECK 2.4.0
3. Create Elasticsearch and Kibana 8.5.1 resources
4. Create an ApmServer 8.5.1 resource with non-root user and group; this should fail to come up, per the linked issue
5. Build the image off this branch (or take a snapshot/BC image), and replace the ApmServer image (`spec.image: ...`); this should work.

## Related issues

Closes https://github.com/elastic/apm-server/issues/9596
